packages/medusa/src/services/batch-job.ts

```diff
@@ -0,0 +1,115 @@
+import { EntityManager } from "typeorm"
+import { BaseService } from "medusa-interfaces"
+import { MedusaError } from "medusa-core-utils"
+
+import { BatchJob } from "../models"
+import { BatchJobRepository } from "../repositories/batch-job"
+import { BatchJobStatus, FilterableBatchJobProps } from "../types/batch-job"
+import { FindConfig } from "../types/common"
+import EventBusService from "./event-bus"
+
+type InjectedContainer = {
+  manager: EntityManager
+  eventBusService: EventBusService
+  batchJobRepository: typeof BatchJobRepository
+}
+
+class BatchJobService extends BaseService<BatchJobService> {
+  protected readonly container_: InjectedContainer
+  protected readonly batchJobRepository_: typeof BatchJobRepository
+  protected readonly eventBus_: EventBusService
+
+  static Events = {
+    CREATED: "batch.created",
+    UPDATED: "batch.updated",
+    COMPLETED: "batch.completed",
+    CANCELED: "batch.canceled",
+    PROCESS_COMPLETE: "batch-process.complete",
+  }
+
+  constructor(container: InjectedContainer) {
```

Comment:
@srindom I though we decided to avoid to inject the container like this to avoid ending up with a service locator? (Event if it is that under the hood after build)